rust/src/dataset.rs

```diff
@@ -369,15 +369,9 @@ impl Dataset {
         }
 
         let mut fragment_id = if matches!(params.mode, WriteMode::Append) {
-            dataset.as_ref().map_or(0, |d| {
-                d.manifest
-                    .fragments
-                    .iter()
-                    .map(|f| f.id)
-                    .max()
-                    .map(|id| id + 1)
-                    .unwrap_or(0)
-            })
+            dataset
+                .as_ref()
+                .map_or(0, |d| d.manifest.max_fragment_id().unwrap_or(0) + 1)
```

Comment:
Good point. I've adjusted the logic for `max_fragment_id()` so that if it is unset (zero) then we compute it based on the fragments in the manifest, like before.